uber/site_sections/summary.py

```diff
@@ -175,12 +175,115 @@ def food_eligible(self, session):
+                    # the export has labels separated by ' / ' and we want to convert that
+                    # back into a comma-separate list of integers
+                    label_lookup = {val: key for key, val in col.type.choices}
+                    vals = [label_lookup[label] for label in val.split(' / ')]
+                    val = ','.join(map(str, vals))
+                elif isinstance(col.type, UTCDateTime):
+                    # we'll need to make sure we use whatever format string we used to
+                    # export this date in the first place
+                    try:
+                        val = UTC.localize(datetime.strptime(val, date_format + ' %H:%M:%S'))
+                    except:
+                        val = UTC.localize(datetime.strptime(val, date_format))
+                elif isinstance(col.type, Date):
+                    val = datetime.strptime(val, date_format).date()
+                elif isinstance(col.type, Integer):
+                    val = int(val)
+
+                # now that we've converted val to whatever it actually needs to be, we
+                # can just set it on the model
+                setattr(model_instance, colname, val)
+
+            try:
+                session.commit()
+            except:
+                log.error('ImportError', exc_info=True)
+                session.rollback()
+                message = 'Import unsuccessful'
+
+            id_list.append(model_instance.id)
+
+        all_instances = session.query(model).filter(model.id.in_(id_list)).all() if id_list else None
+
+        if model == Attendee:
+            form_link = '../registration/form'
+        elif model == Group:
+            form_link = '../groups/form'
+        elif model == Job:
+            form_link = '../jobs/form'
+        else:
+            form_link = None
+
+        return self.csv_import(message, all_instances, form_link)
+
+    def valid_attendees(self):
+        self.export_model(selected_model='attendee')
+
+    def all_attendees(self):
+        self.export_model(selected_model='attendee')
+
+    def csv_export(self, message='', **params):
+        if 'model' in params:
+            self.export_model(selected_model=params['model'])
+
+        return {
+            'message': message,
+            'tables': sorted([model.__name__.lower() for model in Session.all_models()])
+        }
+
     @csv_file
-    def valid_attendees(self, out, session):
-        cols = [getattr(Attendee, col.name) for col in Attendee.__table__.columns]
+    def export_model(self, out, session, selected_model=''):
+        model = session.get_class_by_tablename(selected_model, 'csv_export')
+
+        cols = [getattr(model, col.name) for col in model.__table__.columns]
         out.writerow([col.name for col in cols])
 
-        for attendee in session.valid_attendees().filter(Attendee.first_name != '').order_by(Attendee.badge_num).all():
+        for attendee in session.query(model).all():
```

Comment:
This diff indicates that you DID rename the page in a way that WILL force people to update their bookmarks.  Or am I misreading this?

FWIW, I like renaming things to be more correct.  If we want to not break bookmarks, we can always add a page handler for the old name with a ``raise HTTPRedirect('<new path>')``.